### PR TITLE
add `fmc_url_group`

### LINKED ISSUE #52: Add `fmc_url_group` resource and data source
API: /api/fmc_config/v1/domain/{domainUUID}/object/urlgroups

### PR DESCRIPTION
**Pull request by igiai**
Closes #52

### CODE REVIEW

File: gen/definitions/url_group.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+name: Url Group
```
Comment (danischm) [inline gen/definitions/url_group.yaml:2]: `name: URL Group`

File: gen/definitions/url_group.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+rest_endpoint: /api/fmc_config/v1/domain/{DOMAIN_UUID}/object/urlgroups
+data_source_name_query: true
+doc_category: Objects
+attributes:
+  - model_name: name
+    type: String
+    description: User-created name of the resource.
```
Comment (danischm) [inline gen/definitions/url_group.yaml:9]: Mentioning "resource" might be misleading.. as it is not the name of the TF resource, but rather the object it creates. So maybe something like "User-created name of the object.".

File: gen/definitions/url_group.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  - model_name: description
+    type: String
+    description: Optional user-created description.
+    example: "My URL group"
+  - model_name: overridable
+    type: Bool
+    description: Indicates whether object values can be overridden.
+    exclude_example: true
+    test_value: "true"
+  - model_name: objects
```
Comment (danischm) [inline gen/definitions/url_group.yaml:21]: I would add a `tf_name: urls` here to be more specific of what is expected.

File: gen/definitions/url_group.yaml (hunk 2 above)
Comment (igiai) [reply gen/definitions/url_group.yaml:9]: Sure, makes sense

**Review by danischm — APPROVED** [review]

**Pull request merged**